public-dashboard-client/src/controls/CohortSelect.js

```diff
@@ -0,0 +1,300 @@
+            // keep the clicked item highlighted (highlight is cleared by default)
+            highlightedIndex: state.highlightedIndex,
+          };
+        default:
+          return changes;
+      }
+    },
+    onSelectedItemChange: ({ selectedItem }) => {
+      if (!selectedItem) {
+        return;
+      }
+      let newSelection;
+
+      if (selectedItem.id === SELECT_ALL_ID) {
+        // if all are already selected, deselect all
+        if (selected.length === optionsFromData.length) {
+          newSelection = [];
+        } else {
+          newSelection = [...optionsFromData];
+        }
+      } else {
+        newSelection = [...selected];
+
+        const index = selected.indexOf(selectedItem);
+
+        if (index === -1) {
+          newSelection.push(selectedItem);
+        } else {
+          newSelection.splice(index, 1);
+        }
+        // need to keep selection sorted or labels and colors will get out of sync
+        newSelection.sort((a, b) => ascending(a.label, b.label));
+      }
+
+      setSelected(newSelection);
+    },
+  });
+
+  useEffect(() => {
+    // index 0 is select all and should be ignored here
+    if (highlightedIndex < 1) {
+      onHighlight(undefined);
+    } else {
+      // offset by one due to select all
+      onHighlight(optionsFromData[highlightedIndex - 1]);
+    }
+  }, [highlightedIndex, onHighlight, optionsFromData]);
+
+  const labelProps = getLabelProps();
+  const toggleButtonProps = getToggleButtonProps();
+  return (
+    <>
+      <ControlLabel as="label" {...labelProps}>
+        Cohort
+      </ControlLabel>
+      <ControlValue as="button" type="button" {...toggleButtonProps}>
+        {buttonContents}
+      </ControlValue>
+      <DropdownMenu {...getMenuProps()} as="ul">
+        {isOpen &&
+          visibleOptions.map((opt, index) => {
+            const isSelected = selected.includes(opt);
+            const itemProps = getItemProps({ item: opt, index });
+            return (
+              <DropdownMenuItem
+                {...itemProps}
+                aria-selected={isSelected}
+                as="li"
+                backgroundColor={isSelected ? opt.color : undefined}
+                highlightedSelector={
+                  highlightedIndex === index ? `&#${itemProps.id}` : undefined
+                }
+                key={opt.id}
+              >
+                <MenuItemContents>
+                  {opt.label}
+                  <MenuItemCheckMark src={checkMarkPath} />
+                </MenuItemContents>
+              </DropdownMenuItem>
+            );
+          })}
+      </DropdownMenu>
+    </>
+  );
+}
+
+CustomSelect.propTypes = {
+  buttonContents: PropTypes.node.isRequired,
+  onHighlight: PropTypes.func.isRequired,
+  options: OPTIONS_PROP_TYPE.isRequired,
+  selected: OPTIONS_PROP_TYPE.isRequired,
+  setSelected: PropTypes.func.isRequired,
+};
+
+function NativeSelect({ buttonContents, options, selected, setSelected }) {
```

Comment:
yeah this is a weird one ... the `<select>` is invisible, but on an actual phone os (verified in browserstack, ios as well as android) clicking it will trigger the native UI (the ios carousel or the android dialog). on a desktop there is a little (invisible) scrolling box with all the options on it, you happen to just be clicking the second one over and over because it's under your cursor.

i guess that means there is technically a case where if you make your window really tiny on your desktop, you won't be able to access the menu? for this reason and some others I am not totally thrilled with how this works, either here or in the other dropdowns ... I have not really pushed back on the "fall back to native behavior on mobile" requirement as yet; we thought it would be an easy and graceful fallback but it has turned out to be neither of those things! I don't really think dealing with that would be in scope for this PR but it's worth considering, at the very least as part of the transition to spotlight v1 if not before that